common/src/api/internal/nexus.rs

```diff
@@ -103,3 +103,19 @@ pub struct UpdateArtifact {
 pub enum UpdateArtifactKind {
     Zone,
 }
+
+/// Sent to a sled agent to request the contents of an Instance's serial console.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+pub struct InstanceSerialConsoleRequest {
+    /// Character index in the serial buffer (since instance boot) from which to read.
+    pub byte_offset: Option<isize>,
+    /// Maximum number of bytes of buffered serial console contents (after byte_offset) to return.
+    pub max_bytes: Option<usize>,
+}
+
+/// Contents of an Instance's serial console buffer.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct InstanceSerialConsoleData {
+    pub data: Vec<u8>, // might not be UTF-8.
+    pub last_byte_offset: usize,
```

Comment:
it would return an error if we are requesting a range that *begins* somewhere we don't have cached.

for positive values of `byte_offset`, the returned `last_byte_offset` is effectively `byte_offset + data.len()`, the intent is that the client gets an easy value to plug into the next call's `byte_offset`. (this is more useful when indexing from most-recent, but still potentially convenient when indexing from beginning)